### ISSUE TITLE
The app crashed when I tried to change themes.

### ISSUE THREAD

**Issue by sagorahmed** [ZoidbergZero/Canora-MusicPlayer#8]
**Describe the bug**
When I tried to change the themes to "Mint" "Dark" and "Light", then the app crashed.

**To Reproduce**
1. Open the app.
2. Go to the settings
3. Click on "Themes" and change your theme to dark, mint or light and notice the bug. (Make sure to check multiple times if it works fine)

**Expected behavior**
I expected that I could be able to change the app's themes successfully without getting any trouble.

**Smartphone (please complete the following information):**
 - Device: Samsung a5
 - OS: Android 6.0.1
 - Version: 0.4

**Screenshots**
![bug](https://media.giphy.com/media/SVlIlb0G6R7OYbEkli/giphy.gif)

**LogCat**
I haven't found error logs associated with this app on ADB.

**Comment by ZoidbergZero** [comment]
I cant reproduce this on my System, can you please search the Logcat for "ch.swissproductions.canora" and post the output?

**Comment by sagorahmed** [comment]
```
05-21 17:09:56.990 25038 25052 D libGLESv1: DTS_GLAPI : DTS is not allowed for Package : ch.swissproductions.canora

05-21 17:09:57.030  3164  3445 D NetworkController.MobileSignalController(0/3): onSignalStrengthsChanged signalStrength=SignalStrength: 14 99 -14 -200 -14 -200 -1 99 2147483647 2147483647 2147483647 2147483647 2147483647 0x4 gsm|lte level=4

05-21 17:09:57.030  3164  3445 D NetworkController.MobileSignalController(0/3): getMobileIconGroup(): 9

05-21 17:09:57.040  2697  3591 V WindowStateAnimator: Finishing drawing window Window{409a2fc u0 d0 p25038 ch.swissproductions.canora/ch.swissproductions.canora.MainActivity}: mDrawState=DRAW_PENDING

05-21 17:09:57.040 25038 25038 D ViewRootImpl: MSG_RESIZED_REPORT: ci=Rect(0, 72 - 0, 0) vi=Rect(0, 72 - 0, 0) or=1

05-21 17:09:57.040  2697  2820 D KnoxTimeoutHandler: notifyActivityDrawn [MsgParam] userId: 0 fullscreen is true showWhenlocked is false

05-21 17:09:57.040  2697  2712 D InputMethodManagerService: windowGainedFocus mCurrentFocusedUserId - 0 and mSecureKeypadEnabled-false

05-21 17:09:57.040  2697  2697 D KnoxTimeoutHandler: activityDrawn [MsgParam] userId: 0 fullscreen is true showWhenlocked is false

05-21 17:09:57.040  2697  2820 I ActivityManager: Displayed ch.swissproductions.canora/.MainActivity: +1s214ms

05-21 17:09:57.040  2697  2697 I KnoxTimeoutHandler: SD activityfalse 
```

This is what I found. I don't know this is the exact log or not. If you need more i can give you that logcat file.

**Comment by ZoidbergZero** [comment]
yes attach a full log file of a session where this bug happened to a comment here please

**Comment by sagorahmed** [comment]
Here is the [file](https://drive.google.com/open?id=1GBEbpgkpyfhARhjiH_HDz08lnnV9T5fa)

**Comment by ZoidbergZero** [comment]
I'll have a look, did u already try completely reinstalling the app? Could be a Permissions issue.

**Comment by sagorahmed** [comment]
Yes, I have granted all the permissions it asked. 
Have you shuffled colors in the theme changing process multiple time??

**Comment by ZoidbergZero** [comment]
Try the below version. It could be a problem with the recreate Function which i replaced in this build.
[app-debug.zip](https://github.com/ZoidbergZero/Canora-MusicPlayer/files/3202958/app-debug.zip)

**Comment by sagorahmed** [comment]
No. This issue is still there. :(

**Comment by ZoidbergZero** [comment]
The Error that kills the App is this:

`05-21 19:13:24.655  2245  2245 F DEBUG   : Abort message: 'Encountered EGL error 12291 EGL_BAD_ALLOC during rendering'`

which is a very LowLevel problem which is very strange, as i also cant Reproduce this in my system.

The only other thing i could think of is too much Work on the Main Thread so please try the Build below which doesnt load any files and tell me the Results please. Also what exactly do you mean by "shuffled colors" ?

[app-debug.zip](https://github.com/ZoidbergZero/Canora-MusicPlayer/files/3208244/app-debug.zip)

**Comment by ZoidbergZero** [comment]
See 25d9bb7 i have now Replaced the Settings Dialog with a separate Activity, Can you please try this Build and tell me the Results?

[app-debug.zip](https://github.com/ZoidbergZero/Canora-MusicPlayer/files/3211642/app-debug.zip)

**Comment by sagorahmed** [comment]
> See [25d9bb7](https://github.com/ZoidbergZero/Canora-MusicPlayer/commit/25d9bb7b2c98757395e08e6e1d30c1545300f0c5) i have now Replaced the Settings Dialog with a separate Activity, Can you please try this Build and tell me the Results?
> 
> [app-debug.zip](https://github.com/ZoidbergZero/Canora-MusicPlayer/files/3211642/app-debug.zip)

So sorry for reply lately. 
There is no issue in this build. Both work fine.

**Comment by ZoidbergZero** [comment]
Great that the Issue is solved! Im still wondering what exactly has caused this tho.
I will Rerelease Version 0.4 with the Separate Settings Activity.

**Comment by ZoidbergZero** [comment]
Fixed in Rerelease of v0.4:
https://github.com/ZoidbergZero/Canora-MusicPlayer/releases/tag/v0.4
